jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/resource/NativeImagePathResourceFactory.java

```diff
@@ -0,0 +1,71 @@
+//
+// ========================================================================
+// Copyright (c) 1995-2022 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.util.resource;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.file.FileSystemNotFoundException;
+import java.nio.file.FileSystems;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Collections;
+
+/**
+ * GraalVM Native-Image {@link PathResourceFactory}.
+ */
+public class NativeImagePathResourceFactory extends PathResourceFactory
```

Comment:
Can this include the name `GraalVM` in the beginning?

aka `GraalVMNativeImagePathResourceFactory`

Reply:
I'd very much like to avoid GraalVM specific classes.

This class is could be made generic for any scheme that is supported by Path and a FileSystem if the scheme was parameterized.
So perhaps parameterize the scheme and name it something like `FileSystemResourceFactory`?